File: Common/Messages/Handshake/HandshakeResponseModel.cs
using System;
using IOBootstrap.NET.Common.Messages.Base;

namespace IOBootstrap.NET.Common.Messages.Handshake
{
    public class HandshakeResponseModel : IOResponseModel
    {

        public string KeyID { get; set; }
        public string PublicKeyExponent { get; set; }
        public string PublicKeyModulus { get; set; }

        public HandshakeResponseModel(string publicKeyExponent, string publicKeyModulus, string keyID) : base()
        {
            this.KeyID = keyID;
            this.PublicKeyExponent = publicKeyExponent;
            this.PublicKeyModulus = publicKeyModulus;
        }
    }
}

File: Backoffice.UI/src/screens/main/models/HandshakeResponseModel.ts
import { BaseResponseModel } from "iobootstrap-ui-base";

class HandshakeResponseModel extends BaseResponseModel {

    keyID: string | null;
    publicKeyExponent: string | null;
    publicKeyModulus: string | null;

    constructor() {
        super();

        this.keyID = null;
        this.publicKeyExponent = null;
        this.publicKeyModulus = null;
    }
}

export default HandshakeResponseModel;
